pkg/cfn/builder/cluster.go

```diff
@@ -134,7 +135,13 @@ func (c *ClusterResourceSet) newResource(name string, resource gfn.Resource) *gf
 
 func (c *ClusterResourceSet) addResourcesForControlPlane(subnetDetails *subnetDetails) {
 	clusterVPC := &gfneks.Cluster_ResourcesVpcConfig{
-		SecurityGroupIds: gfnt.NewSlice(c.securityGroups...),
+		EndpointPublicAccess:  gfnt.NewBoolean(*c.spec.VPC.ClusterEndpoints.PublicAccess),
+		EndpointPrivateAccess: gfnt.NewBoolean(*c.spec.VPC.ClusterEndpoints.PrivateAccess),
+		SecurityGroupIds:      gfnt.NewSlice(c.securityGroups...),
+	}
+
+	if cidrs := c.spec.VPC.PublicAccessCIDRs; len(cidrs) > 0 {
```

Comment:
No need for the check. This will just not add anything in case it's empty or nil.

Reply:
Good point.